WooCommerce/WooCommerceTests/ViewRelated/Shipping Label/Create Shipping Label/ShippingLabelPackageDetailsViewModelTests.swift

```diff
@@ -336,6 +336,34 @@ final class ShippingLabelPackageDetailsViewModelTests: XCTestCase {
 
         waitForExpectations(timeout: Constants.expectationTimeout, handler: nil)
     }
+
+    func test_totalWeight_updates_when_selected_package_changes() {
+        // Given
+        let items = [MockOrderItem.sampleItem(name: "Easter Egg", productID: 1, quantity: 1)]
+        let order = MockOrders().makeOrder().copy(siteID: sampleSiteID, items: items)
+        let currencyFormatter = CurrencyFormatter(currencySettings: CurrencySettings())
+
+        // When
+        insert(Product.fake().copy(siteID: sampleSiteID, productID: 1, virtual: false, weight: "120"))
+
+        let viewModel = ShippingLabelPackageDetailsViewModel(order: order,
+                                                             packagesResponse: mockPackageResponse(),
+                                                             selectedPackageID: nil,
+                                                             totalWeight: nil,
+                                                             formatter: currencyFormatter,
+                                                             storageManager: storageManager,
+                                                             weightUnit: "kg")
+
+        // Then
+        XCTAssertEqual(viewModel.totalWeight, "120.0")
+
+        // When
+        viewModel.didSelectPackage("Box")
+        viewModel.confirmPackageSelection()
+
+        // Then
+        XCTAssertEqual(viewModel.totalWeight, "130.0")
+    }
```

Comment:
It would be great if we have more tests for cases when total weight should not be updated automatically:
- when initial total weight is an arbitrary weight (not equal to automatically calculated weight)
- when user enters a new total weight manually.

Reply:
Great ideas! Thanks for keeping me on my toes with test coverage. :) I added those tests in 3dcd3c2.